compiler/rustc_data_structures/src/graph/dominators/mod.rs

```diff
@@ -91,6 +91,10 @@ pub struct Dominators<N: Idx> {
 }
 
 impl<Node: Idx> Dominators<Node> {
+    pub fn dummy() -> Self {
+        Self { post_order_rank: IndexVec::new(), immediate_dominators: IndexVec::new() }
+    }
```

Comment:
Looks like you could `#[derive(Default)]` instead.

Reply:
I wanted to be explicit, maybe there was a reason why there was no empty constructor.

Reply:
For whatever reason, simply deriving doesn't work:

```
error[E0277]: the trait bound `rustc_middle::mir::BasicBlock: std::default::Default` is not satisfied
   --> compiler/rustc_mir/src/borrow_check/mod.rs:290:29
    |
290 |                 dominators: Dominators::default(), // not used
    |                             ^^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `rustc_middle::mir::BasicBlock`
    |
    = note: required because of the requirements on the impl of `std::default::Default` for `Dominators<rustc_middle::mir::BasicBlock>`
    = note: required by `std::default::Default::default`
```